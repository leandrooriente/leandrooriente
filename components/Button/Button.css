.button {
  background: var(--light-gray);
  color: var(--primary-color);
  display: inline-block;
  padding: 10px;
  font-size: 12px;
  text-transform: lowercase;
  text-decoration: none;
}

.button:hover {
  background: var(--highlight-color);
  color: var(--light-gray);
  transition: all var(--fast-transition) ease-in;
}