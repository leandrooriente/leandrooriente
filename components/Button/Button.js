import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import css from './Button.css';

const Button = ({
  className,
  children,
  onClick,
  href,
}) => {
  const classes = `${css.button}${className ? ` ${className}` : ''}`;
  if (onClick) {
    return (
      <button className={classes} onClick={onClick}>
        { children }
      </button>
    );
  }

  if (href) {
    return (
      <Link href={href}>
        <a className={classes}>{ children }</a>
      </Link>
    );
  }

  return null;
};

Button.defaultProps = {
  className: null,
  href: null,
  onClick: null,
};

Button.propTypes = {
  className: PropTypes.string,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
};

export default Button;
