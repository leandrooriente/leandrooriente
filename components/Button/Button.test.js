import React from 'react';
import { mount, shallow } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
  const mockFn = jest.fn();
  it('should render <Button /> component as link when receive href prop', () => {
    const wrapper = mount(
      <Button href="https://test.com">
        test.com
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Button /> component as button when receive onClick prop', () => {
    const wrapper = mount(
      <Button onClick={mockFn}>
        submit
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should return null if missing href and onClick properties', () => {
    const wrapper = shallow(
      <Button>
        null
      </Button>
    );

    expect(wrapper.getElement()).toBe(null);
  });
});
