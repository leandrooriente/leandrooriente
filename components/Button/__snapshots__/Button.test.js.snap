// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render <Button /> component as button when receive onClick prop 1`] = `
<Button
  className={null}
  href={null}
  onClick={[MockFunction]}
>
  <button
    className="button"
    onClick={[MockFunction]}
  >
    submit
  </button>
</Button>
`;

exports[`<Button /> should render <Button /> component as link when receive href prop 1`] = `
<Button
  className={null}
  href="https://test.com"
  onClick={null}
>
  <Link
    href="https://test.com"
  >
    <a
      className="button"
      href="https://test.com"
      onClick={[Function]}
    >
      test.com
    </a>
  </Link>
</Button>
`;
